server/src/main/java/org/elasticsearch/search/aggregations/metrics/HyperLogLogPlusPlus.java

```diff
@@ -485,4 +625,146 @@ public int value() {
             return value;
         }
     }
+
+    private static class LowCardinalityLinearCounting extends AbstractLinearCounting implements Releasable {
+        private final BigArrays bigArrays;
+        private final LowCardinalityLinearCountingIterator iterator;
+        // max number of encoded hashes
+        private final int capacity;
+        // Number of elements stored.
+        private IntArray sizes;
+        // Actual encoded hashes stored.
+        private IntArray hashes;
+        // Defines the position of the data structure. Callers of this object should set this value
+        // before calling any of the methods.
+        protected long bucket;
```

Comment:
Probably something for another time, but I'd love not to have these `bucket` members. The rest of aggregations just passes the bucket around and I'm not 100% sure why we don't so it here. It feels a little confusing.